projects/core/src/lazy-loading/lazy-modules.service.ts

```diff
@@ -114,6 +127,72 @@ export class LazyModulesService implements OnDestroy {
     );
   }
 
+  /**
+   * The purpose of this function is to run MODULE_INITIALIZER logic that can be provided
+   * by a lazy loaded module.  The module is recieved as a function parameter.
+   * This function returns an Observable to the module reference passed as an argument.
+   *
+   * @param {NgModuleRef<any>} moduleRef
+   *
+   * @returns {Observable<NgModuleRef<any>>}
+   */
+
+  public runModuleInitializersForModule(
+    moduleRef: NgModuleRef<any>
+  ): Observable<NgModuleRef<any>> {
+    const moduleInits: any[] = moduleRef.injector.get<any[]>(
+      MODULE_INITIALIZER,
+      [],
+      InjectFlags.Self
+    );
+    const asyncInitPromises = this.runModuleInitializerFunctions(moduleInits);
+    if (asyncInitPromises.length) {
+      return from(Promise.all(asyncInitPromises)).pipe(
+        catchError((error) => {
+          console.error(
+            `MODULE_INITIALIZER promise was rejected while lazy loading a module: `,
+            error
+          );
+          return of(error);
```

Comment:
Wondering if we should not rethrow this error here, as it seems we are swallowing it and the module will be initialized.

Reply:
I added cote to rethrow the error.  It seems to prevent the modules from loading.

I also did that in the eager loading logic.  However, rethrowing the error in the eager loading scenario does not prevent the app from loading.